### ISSUE TITLE
[Outreachy Task Submission] Accessibility issue with "Add Collection" modal form labels

### ISSUE THREAD

**Issue by Chiemezuo** [ushahidi/platform#4889]
# Intro
Accessibility-checking software flags the `Add Collections` modal that opens after clicking on the `Add Collections +` button in the collections modal (that appears after clicking 'ollections' by the side of the screen) because there is no appropriate label for the search field. 

## How to reproduce this issue.
If you use a tool like [WAVE evaluation tool](https://chromewebstore.google.com/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh), you can follow these steps:

1. Open your browser and log into your running Ushahidi deployment (from localhost).
2. Click on the `Collections` button at the top part of the screen. 
3. When the `Collection` modal opens, click on the `Add Collection +` button.
4. Put on the Accessibility browser extension. It will flag the `description` and `collection name` form input fields.

## Why this issue happens
It happens because despite the apparent existence of a label, Angular doesn't do a particularly good job of supporting the `for` attribute.

## Accessibility Violation
Success Criterion 3.3.2 Labels or Instructions

 (Level A)
[Labels](https://www.w3.org/TR/WCAG22/#dfn-labels) or instructions are provided when content requires user input.
